doc/source_debugging.md

```diff
@@ -76,21 +98,21 @@ make -j $(nproc)
 ``` bash
 cd ${WAMR_ROOT}/product-mini/platforms/linux
 mkdir build && cd build
-cmake .. -DWAMR_BUILD_DEBUG_AOT=1
+cmake .. -DCMAKE_BUILD_TYPE:STRING="Debug" -DWAMR_BUILD_DEBUG_AOT=1
```

Comment:
Must we build iwasm with Debug mode?

Reply:
OK, how about we add a note here to mention that the `-DCMAKE_BUILD_TYPE:STRING="Debug"` option is to make lldb be able to showing the source code of iwasm?

Reply:
i guess it makes sense. @maht can you confirm?